src/net/sf/memoranda/ui/AppFrame.java

```diff
@@ -105,7 +107,12 @@ public void actionPerformed(ActionEvent e) {
 
     public Action minimizeAction = new AbstractAction("Close the window") {
         public void actionPerformed(ActionEvent e) {
-            doMinimize();
```

Comment:
Since doMinimize() calls AppFrame.exitNotify() I would suggest calling that before closing/disposing of the window.